src/charm.py

```diff
@@ -110,31 +110,23 @@ def __init__(self, *args):
         self.framework.observe(
             self.on.fiveg_nrf_relation_joined, self._on_fiveg_nrf_relation_joined
         )
-        self.framework.observe(
-            self.on.certificates_relation_created, self._on_certificates_relation_created
-        )
-        self.framework.observe(
-            self.on.certificates_relation_joined, self._on_certificates_relation_joined
-        )
+        self.framework.observe(self.on.certificates_relation_joined, self._configure_nrf)
+        self.framework.observe(self._certificates.on.certificate_available, self._configure_nrf)
         self.framework.observe(
             self.on.certificates_relation_broken, self._on_certificates_relation_broken
         )
-        self.framework.observe(
-            self._certificates.on.certificate_available, self._on_certificate_available
-        )
         self.framework.observe(
             self._certificates.on.certificate_expiring, self._on_certificate_expiring
         )
 
-    def _configure_nrf(self, event: EventBase) -> None:
+    def _configure_nrf(self, event: EventBase) -> None:  # noqa C901
```

Comment:
I would suggest not inhibiting `C901` and finding a refactoring that will solve it instead.

Reply:
I appreciate not adding `#noqa C901`, but I really don't know what would make a _configure() function have fewer if/then statements.  Do I group related ones and make a function call for them?

Reply:
I can see 2 groups of actions : precondition checks and actual configuration actions

Reply:
ok, but essentially does that really remove complexity?  We would need a method to return true/false if the preconditions are met, and stop if not.  Then we need another one for configuration that returns true/false for the restart.  Essentially we are adding complexity with additional logic branches, no?

Reply:
Broke out preconditions